Statistics/Histograms/ThreadsafeInlineHistogram.cs

```diff
@@ -178,6 +178,22 @@ private set
                 _SampleSize = value;
             }
         }
+
+        public IDistributionEnum Type
```

Comment:
What's the purpose of this? just to check what interface is implemented on Threadsafe? It looks like that's static, and I think we can get what interfaces are present on a class through dotnet functions, rather than implementing an enum and property ourselves.

Reply:
it is to be able to check what distribution the IDistribution is, because we have to handle them differently.

Reply:
You can check what distribution an IDistribution is by calling .getType() I think. It should be baked into dotnet. We can discuss more on a webex.